mobile/App.js

```diff
@@ -6,25 +6,111 @@
  * @flow strict-local
  */
 
-import React, {useState} from 'react';
+import React, {useState, useRef, useEffect} from 'react';
 import {SafeAreaView} from 'react-native-safe-area-context';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+
 import HomeView from './src/views/HomeView';
-import ScanView from './src/views/ScanView';
 import VendorView from './src/views/VendorView';
 import RegistrationView from './src/views/RegistrationView';
 import {NavigationContainer} from '@react-navigation/native';
 import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
 import ScreenOption from './src/utilites/ScreenOption';
-import {globalStyleSheet} from './src/utilites/Theme';
 import MyICardView from './src/views/MyICardView';
-
+import { AppState } from "react-native";
 import AuthContext from './src/context/AuthContext';
+import { storeDate } from './src/utilites/StoreDate';
+import { storeUser } from './src/utilites/StoreUser';
 
 const Tab = createBottomTabNavigator();
 
 const App = () => {
   const [user, setUser] = useState(null);
+  const [check, setCheck] = useState(false);
   const value = {user, setUser};
+  const appState = useRef(AppState.currentState);
+  
+  const logoutCheck = async(userData = null,startup = false) => {
+    
+    try{
+        const lastDate = parseInt(await AsyncStorage.getItem('lastOpened'));    
+        if(lastDate != null){
+            //change num in if statement below to a month in milliseconds
+            if(new Date().getTime() - lastDate > 2629800000){
+                setUser(null)
+                storeUser(null)
+            }
+            else{
+                if(startup){
+                  setUser(userData)
```

Comment:
Not sure if this is handled elsewhere, but if the userData has changed in the last 30 days will we be showing them old data or do we have a way to load that new data in our async storage?

Reply:
I didn't want to mess with any fetching of the data stuff since that isn't something I worked on. But as things stand in this code, they'll be automatically set to their old data. They got to then refresh to get the new data (the new task that I'm working on this sprint)

Reply:
In that case we should probably add a TODO to let other developers know this will be address later